salt/modules/glusterfs.py

```diff
@@ -117,11 +133,20 @@ def list_peers():
 
     '''
     root = _gluster_xml('peer status')
-    result = [x.find('hostname').text for x in _iter(root, 'peer')]
-    if len(result) == 0:
+    if not _gluster_ok(root):
         return None
-    else:
-        return result
+
+    result = []
+    for peer in _iter(root, 'peer'):
+        hostname = peer.find('hostname').text
+        hostnames = peer.find('hostnames')
+        if hostnames is not None:
+            for alternative in hostnames:
+                if alternative.text.startswith(hostname + '.'):
+                    hostname = alternative.text
+        result.append(hostname)
+
+    return result
```

Comment:
I think a simple list of consistent peer names would be useful especially in Jinja templates and other simple cases. It could be nice to add another function like peer_status(), which would return a richer set of info as the corresponding Gluster call does.